src/App.tsx

```diff
@@ -17,18 +78,37 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                status={status}
+                onQueryChange={setQuery}
+                onStatusChange={setStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodo={selectedTodo}
+                  onShow={handleShowModal}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          todo={selectedTodo}
+          user={selectedUser}
+          isLoading={isModalLoading}
+          onClose={handleCloseModal}
+        />
+      )}
```

Comment:
Ensure that `selectedTodo` and `selectedUser` are handled properly to avoid potential null reference errors. Consider adding checks or default values where necessary.